fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java

```diff
@@ -1176,20 +1176,14 @@ private void handleKill() throws DdlException {
         if (killCtx == null) {
             ErrorReport.reportDdlException(ErrorCode.ERR_NO_SUCH_THREAD, id);
         }
-        if (context == killCtx) {
-            // Suicide
-            context.setKilled();
-        } else {
+        if (context != killCtx && !killCtx.getQualifiedUser().equals(ConnectContext.get().getQualifiedUser())
+                && !Env.getCurrentEnv().getAccessManager().checkGlobalPriv(ConnectContext.get(),
+                PrivPredicate.ADMIN)) {
             // Check auth
             // Only user itself and user with admin priv can kill connection
-            if (!killCtx.getQualifiedUser().equals(ConnectContext.get().getQualifiedUser())
-                    && !Env.getCurrentEnv().getAccessManager().checkGlobalPriv(ConnectContext.get(),
-                    PrivPredicate.ADMIN)) {
-                ErrorReport.reportDdlException(ErrorCode.ERR_KILL_DENIED_ERROR, id);
-            }
-
-            killCtx.kill(killStmt.isConnectionKill());
+            ErrorReport.reportDdlException(ErrorCode.ERR_KILL_DENIED_ERROR, id);
         }
```

Comment:
why not context.setKilled()?  Is this useless?

Reply:
`killCtx.kill()` contains the `setKill` logic if user wanna kill the conn